### PR TITLE
Use HMCTS Java plugin for code quality tools

### PR DESCRIPTION
**Pull request by banderous**
### Change description ###

Use the [HMCTS Java plugin](https://github.com/hmcts/gradle-java-plugin) to apply Checkstyle, PMD and OWASP dependency checker with HMCTS default settings.

The Java plugin is intended to help teams apply a minimum standard of code quality tools, whilst making it easier for HMCTS to evolve that standard over time. (Eg. we currently have >50 different checkstyle configs in use in HMCTS, to take one example).

This PR does introduce a change to the configuration of the OWASP dependency checker, which now considers runtime dependency configurations only (known non-runtime configurations including tests and checkstyle configurations are excluded).

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -84,21 +86,6 @@ task smoke(type: Test) {
   classpath = sourceSets.smokeTest.runtimeClasspath
 }
 
-checkstyle {
-  maxWarnings = 0
-  toolVersion = '8.29'
-  getConfigDirectory().set(new File(rootDir, 'config/checkstyle'))
-}
-
-pmd {
```
Comment (timja) [inline build.gradle:93]: do we want to keep PMD enabled? sonarqube is supposed to cover it
moving this to the plugin makes it harder to disable or alter the config in a team project I assume?

Comment (banderous) [reply build.gradle:93]: The default PMD settings can be extended with additional rulesets or replaced completely with custom rulesets if desired in buildscript.

Sonarqube runs PMD if I understand correctly? So we'd still want control of the PMD rulesets. Also does sonarqube run analysis locally or just on CI?

Comment (timja) [reply build.gradle:93]: no they've written their own static analysis tool afaik, they used to.

It runs on CI but you can run it locally with sonarlint

Comment (banderous) [reply build.gradle:93]: Re: disabling rulesets. If teams have a need to blanket remove an entire rule then it probably shouldn't be in the default HMCTS rulesets (teams can suppress individual violations where necessary).

For example, I did remove one [PMD rule](https://stackoverflow.com/questions/21592497/java-for-each-loop-being-flagged-as-ur-anomaly-by-pmd) that doesn't work correctly with modern Java constructs like foreach loops.

Comment (doncem) [reply build.gradle:93]: There are a lot of PMD rules which report on "issues" we don't really care. I think by default it should be disabled and have an option to enable and override if necessary

Comment (timja) [reply build.gradle:93]: or just not include it?
@satyachundur ?

Comment (banderous) [reply build.gradle:93]: Agreed we don't want to impose a tool with a bad signal to noise ratio. I'll go through the PMD warnings for a couple of our larger projects and see if it flags up anything useful, and either trim back the ruleset or remove the tool completely.

Comment (banderous) [reply build.gradle:93]: @timja @satyachundur PMD is no longer applied in the java plugin. Are you happy to proceed with this PR as is?

**Review by timja — APPROVED** [review]

**Pull request merged**